keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -663,126 +912,116 @@ together in a discrete set of ways to construct a MIG device with a particular
 "profile". These resources include things like "memory", "number of JPEG
 engines", "discrete memory slices on the physical GPU die", etc.
 
-Using the idea of "sink" and "source" devices, we can construct a full GPU as a
-"source" device with a set of capacities representing the different resources
-in its "bag" of resources. Each MIG device then becomes a "sink" device listing
-out the capacities it requires to build out its specific "profile" and then
-referencing the full GPU it is attached to in its `ConsumesCapacityFrom` field.
+Using the capacity pool to represent the "bag" of capacities, we can define
+a set of devices that pull specific capacity from that pool.
 
-For example, an unrolled "source" device for an NVIDIA A100 GPU looks as follows:
+For example, a capacity pool for an NVIDIA A100 GPU looks as follows:
 ```yaml
-- name: gpu-0
-  composite:
-    attributes:
-      ...
-    capacity:
-      copy-engines:
-        quantity: "7"
-      decoders:
-        quantity: "5"
-      encoders:
-        quantity: "0"
-      jpeg-engines:
-        quantity: "1"
-      memory:
-        quantity: 40Gi
-      memorySlice0:
-        quantity: "1"
-      memorySlice1:
-        quantity: "1"
-      memorySlice2:
-        quantity: "1"
-      memorySlice3:
-        quantity: "1"
-      memorySlice4:
-        quantity: "1"
-      memorySlice5:
-        quantity: "1"
-      memorySlice6:
-        quantity: "1"
-      memorySlice7:
-        quantity: "1"
-      multiprocessors:
-        quantity: "98"
-      ofa-engines:
-        quantity: "1"
+sharedCapacityPools:
+- name: gpu-0-pool
+  capacity:
+    copy-engines: "7"
+    decoders: "5"
+    encoders: "0"
+    jpeg-engines: "1"
+    memory: 40Gi
+    memorySlice0: "1"
+    memorySlice1: "1"
+    memorySlice2: "1"
+    memorySlice3: "1"
+    memorySlice4: "1"
+    memorySlice5: "1"
+    memorySlice6: "1"
+    memorySlice7: "1"
+    multiprocessors: "98"
+    ofa-engines: "1"
 ```
 
-With three example "sink" devices representing MIG partitions defined as follows:
+Three example devices representing MIG partitions can be defined as follows:
 ```yaml
+sharedCapacityPools:
+- name: gpu-0-pool
+  ...
+deviceMixins:
+- name: mig-1g.5gb
+  capacity:
+    copy-engines: "1"
+    decoders: "0"
+    encoders: "0"
+    jpeg-engines: "0"
+    memory: 4864Mi
+    memorySlice0: "1"
+    multiprocessors: "14"
+    ofa-engines: "0"
+  sharedCapacityConsumed:
+  - name: copy-engines
+    capacity: "1"
+  - name: decoders
+    capacity: "0"
+  - name: encoders
+    capacity: "0"
+  - name: jpeg-engines
+    capacity: "0"
+  - name: memory
+    capacity: 4864Mi
+  - name: memorySlice0
+    capacity: "1"
+  - name: multiprocessors
+    capacity: "14"
+  - name: ofa-engines
+    capacity: "0"
```

Comment:
I still can't wrap my head around what it means for a mixin to consume shared capacity from somewhere. Is it not enough to simply list the capacities in the mixing and then when you include a reference to a pool at the point you include the mixin in a deivce, that indicates which pool that capacity is pulled from.

Reply:
Closing this since we have updated this part of the design.